react/game/src/app/screens/Game/components/Board/components/Square/index.js

```diff
@@ -1,9 +1,51 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { string, func } from 'prop-types';
 
 import styles from './styles.scss';
 
-function Square() {
-  return <button className={styles.square}>{Math.random() > 0.5 ? 'X' : 'O'}</button>;
+class Square extends Component {
+  state = {
+    hover: false
+  };
+
+  getValue = () => {
+    if (this.props.value) {
+      return this.props.value;
+    }
+    return this.state.hover && !this.props.winner ? this.props.currentPlayer : '';
+  };
+
+  getSquareClass = () => {
+    const { value } = { ...this.props };
+    if (value === 'X') {
+      return styles.xSelected;
+    } else if (value === 'O') {
+      return styles.oSelected;
+    } else if (!value && !this.props.winner) {
+      return styles.selectableSquare;
+    }
+    return '';
+  };
+
+  render() {
+    return (
+      <button
+        className={`${styles.square} ${this.getSquareClass()}`}
+        onClick={this.props.onClick}
+        onMouseEnter={() => this.setState({ hover: true })}
```

Comment:
Avoid the use of arrow functions in render method. U can check why here https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36

Reply:
awesome.